### ISSUE TITLE
Stop command not working on Pro-C

### ISSUE THREAD

**Issue by bobhavlin** [ecodina/hunter-wifi#8]
Can anyone confirm whether the STOP command is working?  I am able to start zones, but when I try to stop the water keeps flowing and the status returned seems ok.  This is a Hunter Pro-C.

i.e.:

curl http://192.168.2.184/api/start/zone/2?time=2   (starts the zone as expected)

curl http://192.168.2.184/api/stop/zone/2  

{"status": "Request received and processed correctly." ... }

**Comment by anubisg1** [comment]
I use an hunter x-core and stop works just fine for me

https://www.hunterindustries.com/en-metric/irrigation-product/controllers/x-coretm


Also remember that the hunter protocol is a one way communication protocol. Meaning that the message you receive is your esp saying that it received your command and that it sent it out of its pin, but it cannot tell you if the sprinkler got it, nor what the real status is

**Comment by bobhavlin** [comment]
Thanks for confirming it works for you.  With more troubleshooting, I have found that I can start and stop zone 1 as expected, but with any other zones although I can successfully start them (i.e. 2, 3, 4, ... 14) show on the screen "E:RR" when I try to stop them.  Maybe there is a slight difference in the byte sequence among controllers? 

Looking at the [HunterRoam.cpp](https://github.com/ecodina/hunter-wifi/blob/62561864c56152a0d479432be79e6be09c0ff853/esp8266-hunter-sprinkler/lib/HunterRoam/HunterRoam.cpp) it looks like it does a "startZone" with time=0, so that might narrow the problem to lines 177-182 which use time.

**Comment by ecodina** [comment]
I also have a Hunter Pro-C and the last time I tested it I could successfully stop all zones.

Maybe you could check https://github.com/seb821/OpenSprinkler-Firmware-Hunter to see if they also have this issue.

**Comment by bobhavlin** [comment]
Thanks for the response.  I looked at the seb821 code and found the following comment stating:
[_"Hunter protocol doesn't work on bytes, but on combination of 4, 9 and 12 (not used here) bit numbers."_](https://github.com/seb821/OpenSprinkler-Firmware-Hunter/commit/e3598ec7c5ba2801130abb77844c055e29f63aeb?branch=e3598ec7c5ba2801130abb77844c055e29f63aeb&diff=split#r48699612)

I am no expert in the protocol ... or any of this :), but looks like there might be a problem with the  [HunterRoam::startZone](https://github.com/ecodina/hunter-wifi/blob/907ed262c9da7ca07ddc64838196c28de00fcb06/esp8266-hunter-sprinkler/lib/HunterRoam/HunterRoam.cpp#L140) constuction...  Maybe something can be derived from this approach https://github.com/gashtaan/hunter-xcore-firmware  @gashtaan

**Comment by gashtaan** [comment]
@bobhavlin the approach you mentioned is my project of alternative firmware for X-Core units, it's completely different than original one.

Though, I have partially reverse engineered original firmware so I have an idea how it works. I commented seb821's code, but seems he didn't notice it.

**Comment by bobhavlin** [comment]
@gashtaan  Thanks for the comment and reverse engineering!  I was hoping to use your insight in to the protocol for a fix - Any idea why the stop command here might not work in my case?

**Comment by gashtaan** [comment]
@bobhavlin sorry, I have no idea. I don't have any experience with Pro-C units.

X-Core seems to be dumber, it receives the packet from seb821's code OK, but ignores most of it. To start zone on X-Core unit should be enough to send:
11111111/0 10000000/0 ---- 0-------/0 TLTL 1-------/0 THTH 0-------/1 ZZZZ 01111111/

TLTL - time lower nibble
THTH - time upper nibble
ZZZZ - zone number
bits marked '-' are ignored

Note, that this is just a theory... I looked at the code, but never used it or tested it.

**Comment by ecodina** [comment]
@gashtaan Thank you for your input. Although I am an engineer, I do not have an extensive background in electronics. 

According to seb821's code, he supports up to 48 stations. However, from your packet I can see than only 16 are supported. I guess this is a difference between X-Core and Pro-C.

**Comment by gashtaan** [comment]
@ecodina yes, I am aware of this difference, X-Core supports only up to 8 stations at hardware level, so 4 bits in the packet are enough.

Just wanted to point out that X-Core actually parse different part of the packet, so I can't help with Pro-C devices. Maybe there is some compatibility between them and Pro-C understands the simpler packet for stations 1-16, but I'm afraid command number (00000000/1) in the packet does matter and I don't know it.

**Comment by khayamgondal** [comment]
@loopy321 were you able to get the stop command working on pro c?
